src/main/java/eu/europeana/metis/sandbox/domain/DatasetMetadata.java

```diff
@@ -0,0 +1,170 @@
+package eu.europeana.metis.sandbox.domain;
+
+import eu.europeana.metis.sandbox.common.locale.Country;
+import eu.europeana.metis.sandbox.common.locale.Language;
+
+/**
+ * DatasetMetadata class used in {@link eu.europeana.metis.sandbox.service.dataset.DatasetGeneratorService} to provide dataset
+ * metadata to generate it.
+ */
+public class DatasetMetadata {
+  private String datasetId;
+  private String datasetName;
+  private Country country;
+  private Language language;
+
+  private DatasetMetadata(DatasetMetadataBuilder datasetMetadataBuilder) {
+    setDatasetId(datasetMetadataBuilder.datasetId);
```

Comment:
I wouldn't call setters from a constructor, it can behave weirdly in hierarchies.
Also do you actually need them? The purpose of the builder is to avoid using the setter as well.
Perhaps the fields directly and make them final?

Reply:
done, accessing final fields.